
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

<title>SpaceTx: experimental and computational methods compartison for spatially resolved transcriptomics</title>
<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
<link href="https://fonts.googleapis.com/css2?family=Assistant:wght@200;400;600;700&display=swap" rel="stylesheet">
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/style.css" />
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx_highlight.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="author" title="About these documents" href="../about.html" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="SSAM Notebook: BaristaSeq data SSAM analysis" href="nb3.consensus_baristaseq.html" />
    <link rel="prev" title="SSAM Notebook: Multiplexed smFISH data SSAM analysis" href="nb1.consensus_msmfish.html" /> 
  </head><body>
  <nav id="main-navbar" class="navbar navbar-expand-lg navbar-light">
    <div class="container pt-2 pb-4">
      <a class="navbar-brand" href="../index.html"><img id="main-logo" src="/_static/spacetx_logo_web_black.svg"></a>
      <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
    </div>
    <div id="main-menubar" class="container-fluid">
      <div class="container position-relative">
        <div class="collapse navbar-collapse" id="navbarSupportedContent">
          <ul class="navbar-nav me-auto mb-2 mb-lg-0 justify-content-between">
            <li class="nav-item dropdown">
              <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
                About
              </a>
              <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
                <li><a class="dropdown-item" href="../about.html">SpaceTx</a></li>
                <li><a class="dropdown-item" href="../people.html">People</a></li>
                <li><a class="dropdown-item" href="../spacejam.html">SpaceJam Hackathon</a></li>
              </ul>
            </li>
            <li><a class="nav-link" href="../data.html">Data</a></li>
            <li><a class="nav-link" href="../results.html">Results and Analysis</a></li>
            <li><a class="nav-link" href="../publications.html">Publications</a></li>
            <li><a class="nav-link" href="../contribute.html">Contributing Guidelines</a></li>
          </ul>
          <form class="d-flex mb-4" id="main-search" action="../search.html" method="get">
            <input class="form-control form-control-sm me-2" type="search" placeholder="Search" aria-label="Search" name="q">
            <button class="btn btn-success btn-sm" type="submit">Search</button>
          </form>
        </div>
      </div>
    </div>
  </nav>

    <div id="wrapper">
      <div id="main" class="container">
        <div class="sidebar">
        </div>
        <div id="content">
          
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container div.prompt *,
div.nboutput.container div.prompt *,
div.nbinput.container div.input_area pre,
div.nboutput.container div.output_area pre,
div.nbinput.container div.input_area .highlight,
div.nboutput.container div.output_area .highlight {
    border: none;
    padding: 0;
    margin: 0;
    box-shadow: none;
}

div.nbinput.container > div[class*=highlight],
div.nboutput.container > div[class*=highlight] {
    margin: 0;
}

div.nbinput.container div.prompt *,
div.nboutput.container div.prompt * {
    background: none;
}

div.nboutput.container div.output_area .highlight,
div.nboutput.container div.output_area pre {
    background: unset;
}

div.nboutput.container div.output_area div.highlight {
    color: unset;  /* override Pygments text color */
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars and line breaks on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
    white-space: pre;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    /*background: #f5f5f5;*/
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
    margin: 0;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt .copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
.jp-RenderedHTMLCommon table,
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
.jp-RenderedHTMLCommon thead,
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
.jp-RenderedHTMLCommon tr,
.jp-RenderedHTMLCommon th,
.jp-RenderedHTMLCommon td,
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
.jp-RenderedHTMLCommon th,
div.rendered_html th {
  font-weight: bold;
}
.jp-RenderedHTMLCommon tbody tr:nth-child(odd),
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
.jp-RenderedHTMLCommon tbody tr:hover,
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}
</style>
<section id="SSAM-Notebook:-(Allen)-MERFISH-data-SSAM-analysis">
<h1>SSAM Notebook: (Allen) MERFISH data SSAM analysis<a class="headerlink" href="#SSAM-Notebook:-(Allen)-MERFISH-data-SSAM-analysis" title="Permalink to this heading">¶</a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import ssam
import pandas as pd
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.rcParams[&quot;font.family&quot;] = &quot;Arial&quot;
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>cell_class_colors = {
    &quot;Lamp5&quot;: &quot;#DA808C&quot;,
    &quot;Sncg&quot;: &quot;#8510C0&quot;,
    &quot;Serpinf1&quot;: &quot;#8510C0&quot;,
    &quot;Vip&quot;: &quot;#70559A&quot;,
    &quot;Sst&quot;: &quot;#F15A29&quot;,
    &quot;Pvalb&quot;: &quot;#D93137&quot;,
    &quot;L2/3 IT&quot;: &quot;#94D9A1&quot;,
    &quot;L4&quot;: &quot;#00979D&quot;,
    &quot;L5 IT&quot;: &quot;#008A61&quot;,
    &quot;L6 IT&quot;: &quot;#A19922&quot;,
    &quot;L5 PT&quot;: &quot;#0D5B78&quot;,
    &quot;L5 NP&quot;: &quot;#3E9E64&quot;,
    &quot;L6 CT&quot;: &quot;#69A8E6&quot;,
    &quot;L6 PT&quot;: &quot;#69A8E6&quot;, #???
    &quot;L6b&quot;: &quot;#266180&quot;,
    &quot;Meis2&quot;: &quot;#FF0000&quot;,
    &quot;CR&quot;: &quot;#00FF66&quot;,
    &quot;Astro&quot;: &quot;#665C47&quot;,
    &quot;Oligo&quot;: &quot;#53776C&quot;,
    &quot;VLMC&quot;: &quot;#697255&quot;,
    &quot;Peri&quot;: &quot;#665547&quot;,
    &quot;SMC&quot;: &quot;#807059&quot;,
    &quot;Endo&quot;: &quot;#8D6C62&quot;,
    &quot;Macrophage&quot;: &quot;#537358&quot;,
}
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>bad_genes = [&#39;Cd52&#39;, &#39;Mup5&#39;, &#39;Rab3b&#39;, &#39;Rprml&#39;, &#39;Tac2&#39;]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>baysor_spots = pd.read_csv(&quot;data/baysor/merfish/segmentation.csv&quot;)
spots = pd.read_csv(&quot;data/raw/Allen_MERFISH_spots_with_anatomy.csv&quot;)
spots = spots.drop(spots.columns[:2], axis=1).drop_duplicates().reset_index(drop=True)
spots[&#39;is_noise&#39;] = baysor_spots.is_noise
spots[&#39;cell&#39;] = baysor_spots.cell
spots = spots[spots.layer != &#39;outside_VISp&#39;]
spots = spots[~spots.gene.str.contains(&quot;Blank-&quot;)]
for bad_gene in bad_genes:
    spots = spots[spots.gene != bad_gene]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>coordinates_visp = [[2524.8618583932202, 6576.111948820892],
   [1688.0005423837813, 5869.404423590317],
   [2299.438470759311, 4923.428032738741],
   [3273.6610578379136, 5452.419069146657],
   [2851.6245044166735, 6284.883773573806],
   [2851.6245044166735, 6284.883773573806],
   [2524.8618583932202, 6576.111948820892]]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import json
from shapely.geometry import Point, Polygon
p = Polygon(coordinates_visp)
spots[&#39;VISp&#39;] = [True if p.intersects(Point(a)) else False for a in spots[[&quot;x_um&quot;,&quot;y_um&quot;]].values]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>spots = spots[spots[&#39;VISp&#39;]]
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[20, 20])
plt.plot(*p.exterior.xy, c=&quot;r&quot;)
plt.scatter(spots.x_um, spots.y_um, s=0.1)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;matplotlib.collections.PathCollection at 0x7f53e95ea910&gt;
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_10_1.png" src="../_images/results_nb2.consensus_merfish_10_1.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>beta = 1.03388671635385
rotm = np.array([[np.cos(beta), np.sin(beta)], [-np.sin(beta), np.cos(beta)]])
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>pos_um = np.array([spots.x_um, spots.y_um])
rot_um = np.dot(pos_um.T, rotm)
rot_um[:, 0] -= np.min(rot_um[:, 0])
rot_um[:, 1] -= np.min(rot_um[:, 1])
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[15, 15])
plt.scatter(*rot_um.T, s=1)
plt.xlim([0, 1350])
plt.ylim([0, 1350])
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
(0.0, 1350.0)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_13_1.png" src="../_images/results_nb2.consensus_merfish_13_1.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>ds = ssam.SSAMDataset(&quot;ssam_data/allen_merfish&quot;)
analysis = ssam.SSAMAnalysis(ds, ncores=10, verbose=True)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>locations = spots[[&quot;gene&quot;, &quot;x_um&quot;, &quot;y_um&quot;, &quot;cell&quot;]].rename(columns={&#39;x_um&#39;: &#39;x&#39;, &#39;y_um&#39;: &#39;y&#39;}).set_index(&#39;gene&#39;)
locations.x = rot_um[:, 0]
locations.y = rot_um[:, 1]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>locations.to_csv(&quot;Allen_MERFISH_filtered.csv&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.run_kde(locations=locations, width=locations.x.max(), height=locations.y.max(), re_run=True)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Running KDE for gene 1700022I11Rik...
Saving KDE for gene 1700022I11Rik...
Running KDE for gene 1810046K07Rik...
Saving KDE for gene 1810046K07Rik...
Running KDE for gene 5031425F14Rik...
Saving KDE for gene 5031425F14Rik...
Running KDE for gene 5730522E02Rik...
Saving KDE for gene 5730522E02Rik...
Running KDE for gene Acta2...
Saving KDE for gene Acta2...
Running KDE for gene Adam2...
Saving KDE for gene Adam2...
Running KDE for gene Adamts2...
Saving KDE for gene Adamts2...
Running KDE for gene Adamts4...
Saving KDE for gene Adamts4...
Running KDE for gene Adra1b...
Saving KDE for gene Adra1b...
Running KDE for gene Alk...
Saving KDE for gene Alk...
Running KDE for gene Ankfn1...
Saving KDE for gene Ankfn1...
Running KDE for gene Ano4...
Saving KDE for gene Ano4...
Running KDE for gene Aqp4...
Saving KDE for gene Aqp4...
Running KDE for gene Asic4...
Saving KDE for gene Asic4...
Running KDE for gene B4galnt2...
Saving KDE for gene B4galnt2...
Running KDE for gene B4galnt3...
Saving KDE for gene B4galnt3...
Running KDE for gene Barx2...
Saving KDE for gene Barx2...
Running KDE for gene Bcl11b...
Saving KDE for gene Bcl11b...
Running KDE for gene Bdnf...
Saving KDE for gene Bdnf...
Running KDE for gene Bgn...
Saving KDE for gene Bgn...
Running KDE for gene Blnk...
Saving KDE for gene Blnk...
Running KDE for gene Bmpr1b...
Saving KDE for gene Bmpr1b...
Running KDE for gene Brinp3...
Saving KDE for gene Brinp3...
Running KDE for gene C1ql3...
Saving KDE for gene C1ql3...
Running KDE for gene C1qtnf7...
Saving KDE for gene C1qtnf7...
Running KDE for gene Cacng5...
Saving KDE for gene Cacng5...
Running KDE for gene Calb1...
Saving KDE for gene Calb1...
Running KDE for gene Calb2...
Saving KDE for gene Calb2...
Running KDE for gene Camk2d...
Saving KDE for gene Camk2d...
Running KDE for gene Car3...
Saving KDE for gene Car3...
Running KDE for gene Cbln2...
Saving KDE for gene Cbln2...
Running KDE for gene Cbln4...
Saving KDE for gene Cbln4...
Running KDE for gene Ccbe1...
Saving KDE for gene Ccbe1...
Running KDE for gene Ccdc162...
Saving KDE for gene Ccdc162...
Running KDE for gene Ccdc3...
Saving KDE for gene Ccdc3...
Running KDE for gene Ccdc80...
Saving KDE for gene Ccdc80...
Running KDE for gene Ccnb1...
Saving KDE for gene Ccnb1...
Running KDE for gene Cd14...
Saving KDE for gene Cd14...
Running KDE for gene Cd24a...
Saving KDE for gene Cd24a...
Running KDE for gene Cdca7...
Saving KDE for gene Cdca7...
Running KDE for gene Cdcp1...
Saving KDE for gene Cdcp1...
Running KDE for gene Cdh12...
Saving KDE for gene Cdh12...
Running KDE for gene Cdh13...
Saving KDE for gene Cdh13...
Running KDE for gene Cdh20...
Saving KDE for gene Cdh20...
Running KDE for gene Cdh9...
Saving KDE for gene Cdh9...
Running KDE for gene Ceacam9...
Saving KDE for gene Ceacam9...
Running KDE for gene Cemip...
Saving KDE for gene Cemip...
Running KDE for gene Chat...
Saving KDE for gene Chat...
Running KDE for gene Chn2...
Saving KDE for gene Chn2...
Running KDE for gene Chodl...
Saving KDE for gene Chodl...
Running KDE for gene Chrm2...
Saving KDE for gene Chrm2...
Running KDE for gene Chrna2...
Saving KDE for gene Chrna2...
Running KDE for gene Cldn5...
Saving KDE for gene Cldn5...
Running KDE for gene Clrn1...
Saving KDE for gene Clrn1...
Running KDE for gene Cnr1...
Saving KDE for gene Cnr1...
Running KDE for gene Cntnap5b...
Saving KDE for gene Cntnap5b...
Running KDE for gene Cobll1...
Saving KDE for gene Cobll1...
Running KDE for gene Col14a1...
Saving KDE for gene Col14a1...
Running KDE for gene Col15a1...
Saving KDE for gene Col15a1...
Running KDE for gene Col23a1...
Saving KDE for gene Col23a1...
Running KDE for gene Col24a1...
Saving KDE for gene Col24a1...
Running KDE for gene Col25a1...
Saving KDE for gene Col25a1...
Running KDE for gene Corin...
Saving KDE for gene Corin...
Running KDE for gene Cplx3...
Saving KDE for gene Cplx3...
Running KDE for gene Crhr2...
Saving KDE for gene Crhr2...
Running KDE for gene Crispld2...
Saving KDE for gene Crispld2...
Running KDE for gene Cspg4...
Saving KDE for gene Cspg4...
Running KDE for gene Ctss...
Saving KDE for gene Ctss...
Running KDE for gene Cux2...
Saving KDE for gene Cux2...
Running KDE for gene Cxcl14...
Saving KDE for gene Cxcl14...
Running KDE for gene Daam2...
Saving KDE for gene Daam2...
Running KDE for gene Dmkn...
Saving KDE for gene Dmkn...
Running KDE for gene Dnase1l3...
Saving KDE for gene Dnase1l3...
Running KDE for gene Dscaml1...
Saving KDE for gene Dscaml1...
Running KDE for gene Egfem1...
Saving KDE for gene Egfem1...
Running KDE for gene Egfr...
Saving KDE for gene Egfr...
Running KDE for gene Egln3...
Saving KDE for gene Egln3...
Running KDE for gene Egr2...
Saving KDE for gene Egr2...
Running KDE for gene Elfn1...
Saving KDE for gene Elfn1...
Running KDE for gene Enpp6...
Saving KDE for gene Enpp6...
Running KDE for gene Epha7...
Saving KDE for gene Epha7...
Running KDE for gene Fam19a2...
Saving KDE for gene Fam19a2...
Running KDE for gene Fam84b...
Saving KDE for gene Fam84b...
Running KDE for gene Fbxl7...
Saving KDE for gene Fbxl7...
Running KDE for gene Fezf2...
Saving KDE for gene Fezf2...
Running KDE for gene Flrt3...
Saving KDE for gene Flrt3...
Running KDE for gene Flt1...
Saving KDE for gene Flt1...
Running KDE for gene Fndc7...
Saving KDE for gene Fndc7...
Running KDE for gene Fosb...
Saving KDE for gene Fosb...
Running KDE for gene Foxp2...
Saving KDE for gene Foxp2...
Running KDE for gene Frem2...
Saving KDE for gene Frem2...
Running KDE for gene Fst...
Saving KDE for gene Fst...
Running KDE for gene Gad1...
Saving KDE for gene Gad1...
Running KDE for gene Gad2...
Saving KDE for gene Gad2...
Running KDE for gene Gfap...
Saving KDE for gene Gfap...
Running KDE for gene Glra1...
Saving KDE for gene Glra1...
Running KDE for gene Gpc6...
Saving KDE for gene Gpc6...
Running KDE for gene Grik1...
Saving KDE for gene Grik1...
Running KDE for gene Grin3a...
Saving KDE for gene Grin3a...
Running KDE for gene Grm1...
Saving KDE for gene Grm1...
Running KDE for gene Grm8...
Saving KDE for gene Grm8...
Running KDE for gene Hpse...
Saving KDE for gene Hpse...
Running KDE for gene Hs3st5...
Saving KDE for gene Hs3st5...
Running KDE for gene Igf2...
Saving KDE for gene Igf2...
Running KDE for gene Igfbp4...
Saving KDE for gene Igfbp4...
Running KDE for gene Igfbp5...
Saving KDE for gene Igfbp5...
Running KDE for gene Igfbp6...
Saving KDE for gene Igfbp6...
Running KDE for gene Ikzf2...
Saving KDE for gene Ikzf2...
Running KDE for gene Il1rapl2...
Saving KDE for gene Il1rapl2...
Running KDE for gene Il4ra...
Saving KDE for gene Il4ra...
Running KDE for gene Inpp4b...
Saving KDE for gene Inpp4b...
Running KDE for gene Iqgap2...
Saving KDE for gene Iqgap2...
Running KDE for gene Itgb8...
Saving KDE for gene Itgb8...
Running KDE for gene Kcng1...
Saving KDE for gene Kcng1...
Running KDE for gene Kcnj8...
Saving KDE for gene Kcnj8...
Running KDE for gene L3mbtl4...
Saving KDE for gene L3mbtl4...
Running KDE for gene Lama3...
Saving KDE for gene Lama3...
Running KDE for gene Lamp5...
Saving KDE for gene Lamp5...
Running KDE for gene Lhx6...
Saving KDE for gene Lhx6...
Running KDE for gene Lmo1...
Saving KDE for gene Lmo1...
Running KDE for gene Lsp1...
Saving KDE for gene Lsp1...
Running KDE for gene Ltf...
Saving KDE for gene Ltf...
Running KDE for gene Luzp2...
Saving KDE for gene Luzp2...
Running KDE for gene Lypd1...
Saving KDE for gene Lypd1...
Running KDE for gene Lyzl4...
Saving KDE for gene Lyzl4...
Running KDE for gene March1...
Saving KDE for gene March1...
Running KDE for gene Marcksl1...
Saving KDE for gene Marcksl1...
Running KDE for gene Meis2...
Saving KDE for gene Meis2...
Running KDE for gene Moxd1...
Saving KDE for gene Moxd1...
Running KDE for gene Mrc1...
Saving KDE for gene Mrc1...
Running KDE for gene Mrgprx2...
Saving KDE for gene Mrgprx2...
Running KDE for gene Muc20...
Saving KDE for gene Muc20...
Running KDE for gene Myh14...
Saving KDE for gene Myh14...
Running KDE for gene Ndst4...
Saving KDE for gene Ndst4...
Running KDE for gene Nhs...
Saving KDE for gene Nhs...
Running KDE for gene Nkain3...
Saving KDE for gene Nkain3...
Running KDE for gene Nnmt...
Saving KDE for gene Nnmt...
Running KDE for gene Nos1...
Saving KDE for gene Nos1...
Running KDE for gene Npas1...
Saving KDE for gene Npas1...
Running KDE for gene Npnt...
Saving KDE for gene Npnt...
Running KDE for gene Npsr1...
Saving KDE for gene Npsr1...
Running KDE for gene Npy2r...
Saving KDE for gene Npy2r...
Running KDE for gene Nr2f2...
Saving KDE for gene Nr2f2...
Running KDE for gene Nr4a1...
Saving KDE for gene Nr4a1...
Running KDE for gene Nr4a2...
Saving KDE for gene Nr4a2...
Running KDE for gene Ntng2...
Saving KDE for gene Ntng2...
Running KDE for gene Nxph1...
Saving KDE for gene Nxph1...
Running KDE for gene Nxph2...
Saving KDE for gene Nxph2...
Running KDE for gene Nxph4...
Saving KDE for gene Nxph4...
Running KDE for gene Olah...
Saving KDE for gene Olah...
Running KDE for gene Olfm3...
Saving KDE for gene Olfm3...
Running KDE for gene Opalin...
Saving KDE for gene Opalin...
Running KDE for gene Oprk1...
Saving KDE for gene Oprk1...
Running KDE for gene Osr1...
Saving KDE for gene Osr1...
Running KDE for gene Otof...
Saving KDE for gene Otof...
Running KDE for gene Parm1...
Saving KDE for gene Parm1...
Running KDE for gene Pcdh8...
Saving KDE for gene Pcdh8...
Running KDE for gene Pde11a...
Saving KDE for gene Pde11a...
Running KDE for gene Pdgfc...
Saving KDE for gene Pdgfc...
Running KDE for gene Pdgfra...
Saving KDE for gene Pdgfra...
Running KDE for gene Pdlim5...
Saving KDE for gene Pdlim5...
Running KDE for gene Penk...
Saving KDE for gene Penk...
Running KDE for gene Phactr2...
Saving KDE for gene Phactr2...
Running KDE for gene Plch1...
Saving KDE for gene Plch1...
Running KDE for gene Plcxd3...
Saving KDE for gene Plcxd3...
Running KDE for gene Pld5...
Saving KDE for gene Pld5...
Running KDE for gene Plekhg3...
Saving KDE for gene Plekhg3...
Running KDE for gene Pou3f1...
Saving KDE for gene Pou3f1...
Running KDE for gene Pou3f3...
Saving KDE for gene Pou3f3...
Running KDE for gene Pou6f2...
Saving KDE for gene Pou6f2...
Running KDE for gene Prdm8...
Saving KDE for gene Prdm8...
Running KDE for gene Prok2...
Saving KDE for gene Prok2...
Running KDE for gene Prokr2...
Saving KDE for gene Prokr2...
Running KDE for gene Prox1...
Saving KDE for gene Prox1...
Running KDE for gene Prr16...
Saving KDE for gene Prr16...
Running KDE for gene Prss12...
Saving KDE for gene Prss12...
Running KDE for gene Prss23...
Saving KDE for gene Prss23...
Running KDE for gene Ptger3...
Saving KDE for gene Ptger3...
Running KDE for gene Ptprk...
Saving KDE for gene Ptprk...
Running KDE for gene Ptprm...
Saving KDE for gene Ptprm...
Running KDE for gene Ptprt...
Saving KDE for gene Ptprt...
Running KDE for gene Ptpru...
Saving KDE for gene Ptpru...
Running KDE for gene Pvalb...
Saving KDE for gene Pvalb...
Running KDE for gene Pxdc1...
Saving KDE for gene Pxdc1...
Running KDE for gene Ramp1...
Saving KDE for gene Ramp1...
Running KDE for gene Reln...
Saving KDE for gene Reln...
Running KDE for gene Rerg...
Saving KDE for gene Rerg...
Running KDE for gene Rfx3...
Saving KDE for gene Rfx3...
Running KDE for gene Rgs5...
Saving KDE for gene Rgs5...
Running KDE for gene Rgs6...
Saving KDE for gene Rgs6...
Running KDE for gene Rnf152...
Saving KDE for gene Rnf152...
Running KDE for gene Ror1...
Saving KDE for gene Ror1...
Running KDE for gene Rorb...
Saving KDE for gene Rorb...
Running KDE for gene Rspo1...
Saving KDE for gene Rspo1...
Running KDE for gene Rxfp1...
Saving KDE for gene Rxfp1...
Running KDE for gene Rxfp2...
Saving KDE for gene Rxfp2...
Running KDE for gene Satb2...
Saving KDE for gene Satb2...
Running KDE for gene Scgn...
Saving KDE for gene Scgn...
Running KDE for gene Sema3e...
Saving KDE for gene Sema3e...
Running KDE for gene Sema5a...
Saving KDE for gene Sema5a...
Running KDE for gene Serpinf1...
Saving KDE for gene Serpinf1...
Running KDE for gene Sertm1...
Saving KDE for gene Sertm1...
Running KDE for gene Sgcd...
Saving KDE for gene Sgcd...
Running KDE for gene Shisa9...
Saving KDE for gene Shisa9...
Running KDE for gene Slc17a6...
Saving KDE for gene Slc17a6...
Running KDE for gene Slc17a7...
Saving KDE for gene Slc17a7...
Running KDE for gene Slc17a8...
Saving KDE for gene Slc17a8...
Running KDE for gene Slc25a13...
Saving KDE for gene Slc25a13...
Running KDE for gene Slc30a3...
Saving KDE for gene Slc30a3...
Running KDE for gene Slc32a1...
Saving KDE for gene Slc32a1...
Running KDE for gene Slc44a5...
Saving KDE for gene Slc44a5...
Running KDE for gene Slco5a1...
Saving KDE for gene Slco5a1...
Running KDE for gene Sncg...
Saving KDE for gene Sncg...
Running KDE for gene Sox10...
Saving KDE for gene Sox10...
Running KDE for gene Sox6...
Saving KDE for gene Sox6...
Running KDE for gene Sp8...
Saving KDE for gene Sp8...
Running KDE for gene Spon1...
Saving KDE for gene Spon1...
Running KDE for gene Sst...
Saving KDE for gene Sst...
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.load_kde()
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[15, 15])
ds.plot_l1norm(rotate=3)
plt.xlim
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;function matplotlib.pyplot.xlim(*args, **kwargs)&gt;
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_19_1.png" src="../_images/results_nb2.consensus_merfish_19_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.find_localmax()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Found 4535 local max vectors.
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[15, 15])
ds.plot_l1norm(rotate=3, cmap=&quot;Greys&quot;)
ds.plot_localmax(rotate=3)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_21_0.png" src="../_images/results_nb2.consensus_merfish_21_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.normalize_vectors()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Loaded a cached normalized vector field (to avoid this behavior, set re_run=True).
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>cell_by_gene = pd.read_csv(&quot;data/jeremy_filtered/MERFISH_v1_filtered_cellxgene.csv&quot;)
cell_by_gene = cell_by_gene.set_index(&#39;gene_name&#39;).T[ds.genes]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from sklearn.preprocessing import normalize, scale
#cell_by_gene_normalized = ssam.run_sctransform(cell_by_gene.reset_index(drop=True), plot_model_pars=True)[0]
cell_by_gene_normalized = np.log(normalize(cell_by_gene, norm=&quot;l1&quot;, axis=1) * 10 + 1)
cell_by_gene_normalized_scaled = scale(cell_by_gene_normalized)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from collections import defaultdict
from itertools import chain

def sort_genes(centroids, tbl, genes, min_exp=0.5):
    sorted_genes = defaultdict(lambda: [])
    sorted_cnt = 0
    while sorted_cnt &lt; len(genes):
        for cidx, mean_cl in enumerate(centroids):
            for gidx in np.argsort(mean_cl)[::-1]:
                if all([not genes[gidx] in l for l in sorted_genes.values()]):
                    if mean_cl[gidx] &lt; min_exp:
                        sorted_genes[&quot;rem&quot;].append(genes[gidx])
                    else:
                        sorted_genes[cidx].append(genes[gidx])
                    sorted_cnt += 1
                    break
    sorted_genes = list(chain(*[sorted_genes[i] for i in range(len(centroids))])) + sorted_genes[&quot;rem&quot;]
    sorted_gidx = [list(genes).index(g) for g in sorted_genes]
    return tbl[:, sorted_gidx], sorted_genes
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>def plot_heatmap(sorted_cbg, sorted_genes, calls, uniq_calls, cols, figsize):
    from sklearn import preprocessing
    from mpl_toolkits.axes_grid1 import Divider, Size
    from matplotlib import patches

    rects = []
    sorted_cbg2 = np.zeros_like(sorted_cbg)
    curpos = 0
    for idx, (cell_type, col) in enumerate(zip(uniq_calls, cols)):
        cl_vecs = sorted_cbg[calls.subclass == cell_type]
        sorted_cbg2[curpos:curpos+len(cl_vecs)] = cl_vecs
        rects.append(patches.Rectangle((curpos, 0), curpos+len(cl_vecs), 1, linewidth=0, facecolor=col))
        curpos += len(cl_vecs)


    fig = plt.figure(figsize=figsize)
    #fig, axes = plt.subplots(2, 1, figsize=[20, 10], sharex=True)

    h = [Size.Fixed(1.5), Size.Scaled(1.0)]
    v = [Size.Fixed(0), Size.Scaled(1.0), Size.Fixed(0.05), Size.Fixed(0.3)]
    divider = Divider(fig, (0, 0, 1, 1), h, v, aspect=False)

    ax_heatmap = fig.add_axes(divider.get_position(), axes_locator=divider.new_locator(nx=1, ny=1))
    ax_ctbar = fig.add_axes(divider.get_position(), axes_locator=divider.new_locator(nx=1, ny=3), sharex=ax_heatmap)

    for rect in rects:
        ax_ctbar.add_patch(rect)

    ax_ctbar.axes.xaxis.set_visible(False)
    ax_ctbar.axes.yaxis.set_visible(False)
    for sp in ax_ctbar.spines.values():
        sp.set_linewidth(0.5)
        sp.set_color(&#39;k&#39;)

    sns.heatmap(sorted_cbg2.T[::-1, :], vmin=-2.5, vmax=2.5, cmap=&#39;bwr&#39;, yticklabels=sorted_genes[::-1],
                cbar=None, ax=ax_heatmap)
    ax_heatmap.axes.xaxis.set_visible(False)
    for tick in ax_heatmap.get_yticklabels():
        tick.set_fontname(&quot;Arial&quot;)
    for sp in ax_heatmap.spines.values():
        sp.set_linewidth(0.5)
        sp.set_color(&#39;k&#39;)
        sp.set_visible(True)
    plt.yticks(rotation=0)

    #ax_hist = fig.add_axes([1.02, 0.74, 0.08, 0.1])
    #ax_hist.hist(np.ravel(sorted_cbg2), bins=100, histtype=&#39;step&#39;, lw=3, color=&#39;lime&#39;)
    #ax_hist.set_xlim([-2.5, 2.5])
    #ax_hist.axes.xaxis.set_ticks([-2.5, 0, 2.5])
    #ax_hist.axes.yaxis.set_visible(False)

    return fig
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>calls_nwcs = pd.read_csv(&quot;consensus_calls/renee/MERFISH_v1_filtered_combined_mapping_neg_weight_subclass.csv&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>calls_nwcs
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_name</th>
      <th>subclass</th>
      <th>rmax</th>
      <th>x</th>
      <th>y</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>L6 PT</td>
      <td>0.494448</td>
      <td>712.40142</td>
      <td>204.819190</td>
    </tr>
    <tr>
      <th>1</th>
      <td>12</td>
      <td>L2/3 IT</td>
      <td>0.682338</td>
      <td>1217.39318</td>
      <td>883.516611</td>
    </tr>
    <tr>
      <th>2</th>
      <td>18</td>
      <td>L2/3 IT</td>
      <td>0.638946</td>
      <td>891.96394</td>
      <td>1036.272708</td>
    </tr>
    <tr>
      <th>3</th>
      <td>20</td>
      <td>L2/3 IT</td>
      <td>0.367690</td>
      <td>654.74648</td>
      <td>973.435156</td>
    </tr>
    <tr>
      <th>4</th>
      <td>22</td>
      <td>L2/3 IT</td>
      <td>0.771932</td>
      <td>645.90474</td>
      <td>923.882162</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>2145</th>
      <td>5873</td>
      <td>L6 PT</td>
      <td>0.712868</td>
      <td>1172.87814</td>
      <td>136.951414</td>
    </tr>
    <tr>
      <th>2146</th>
      <td>5875</td>
      <td>L2/3 IT</td>
      <td>0.497523</td>
      <td>241.48159</td>
      <td>665.555457</td>
    </tr>
    <tr>
      <th>2147</th>
      <td>5877</td>
      <td>L4</td>
      <td>0.568966</td>
      <td>968.05080</td>
      <td>710.759896</td>
    </tr>
    <tr>
      <th>2148</th>
      <td>5878</td>
      <td>Pvalb</td>
      <td>0.649862</td>
      <td>859.54284</td>
      <td>480.890285</td>
    </tr>
    <tr>
      <th>2149</th>
      <td>5990</td>
      <td>L2/3 IT</td>
      <td>0.461098</td>
      <td>482.98299</td>
      <td>1056.178312</td>
    </tr>
  </tbody>
</table>
<p>2150 rows × 5 columns</p>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>uniq_celltypes_nwcs = [cl for cl in cell_class_colors.keys() if cl in calls_nwcs.subclass.unique()]
centroids_nwcs = []
for cell_type in uniq_celltypes_nwcs:
    centroids_nwcs.append(np.mean(cell_by_gene_normalized[calls_nwcs.subclass == cell_type], axis=0))
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>centroids_scaled_nwcs = []
for cell_type in uniq_celltypes_nwcs:
    centroids_scaled_nwcs.append(np.mean(cell_by_gene_normalized_scaled[calls_nwcs.subclass == cell_type], axis=0))

sorted_cbg, sorted_genes = sort_genes(centroids_scaled_nwcs, cell_by_gene_normalized_scaled, ds.genes)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>cols = [cell_class_colors[ct] for ct in uniq_celltypes_nwcs]
plot_heatmap(sorted_cbg[:, ::-1], sorted_genes[::-1], calls_nwcs, uniq_celltypes_nwcs, cols, [20, 50]).savefig(&quot;merfish_heatmap_nwcs.pdf&quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_31_0.png" src="../_images/results_nb2.consensus_merfish_31_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[156]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from sklearn import preprocessing
fig, axes = plt.subplots(len(uniq_celltypes_nwcs), 1, figsize=[20, len(uniq_celltypes_nwcs)*2])
plt.subplots_adjust(hspace=0)
for idx, cell_type in enumerate(uniq_celltypes_nwcs):
    cl_vecs = sorted_cbg[calls_nwcs.subclass == cell_type]
    if len(cl_vecs) == 1:
        cl_vecs = np.array([cl_vecs[0], cl_vecs[0]])
    sns.violinplot(ax=axes[idx], data=cl_vecs[:, -30:], width=1, cut=0)
    axes[idx].set_ylabel(cell_type)
    axes[idx].set_yticks([])
    axes[idx].set_ylim([-5, 5])
axes[idx].set_xticklabels(sorted_genes[-30:], rotation=90)
pass
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_32_0.png" src="../_images/results_nb2.consensus_merfish_32_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>map_colors_nwcs = [cell_class_colors[ct] for ct in uniq_celltypes_nwcs]
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">&lt;ipython-input-11-bb2957d2c7e3&gt;</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>map_colors_nwcs <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">[</span>cell_class_colors<span class="ansi-blue-fg">[</span>ct<span class="ansi-blue-fg">]</span> <span class="ansi-green-fg">for</span> ct <span class="ansi-green-fg">in</span> uniq_celltypes_nwcs<span class="ansi-blue-fg">]</span>

<span class="ansi-green-fg">&lt;ipython-input-11-bb2957d2c7e3&gt;</span> in <span class="ansi-cyan-fg">&lt;listcomp&gt;</span><span class="ansi-blue-fg">(.0)</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>map_colors_nwcs <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">[</span>cell_class_colors<span class="ansi-blue-fg">[</span>ct<span class="ansi-blue-fg">]</span> <span class="ansi-green-fg">for</span> ct <span class="ansi-green-fg">in</span> uniq_celltypes_nwcs<span class="ansi-blue-fg">]</span>

<span class="ansi-red-fg">NameError</span>: name &#39;cell_class_colors&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.map_celltypes(centroids_nwcs)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Generating cell-type map for centroid #0...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #1...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #2...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #3...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #4...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #5...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #6...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #7...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #8...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #9...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #10...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #11...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #12...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #13...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #14...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #15...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #16...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #17...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.filter_celltypemaps(min_norm=0.6, min_r=0.6)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[20, 20])
ds.plot_celltypes_map(rotate=3, colors=map_colors_nwcs)
plt.title(&quot;MERFISH - NWCS (SSAM)&quot;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 1.0, &#39;MERFISH - NWCS (SSAM)&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_36_1.png" src="../_images/results_nb2.consensus_merfish_36_1.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>ds.centroids = centroids_nwcs # TODO: this should not be necessary!
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.bin_celltypemaps(step=10, radius=100)
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[37]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.find_domains(n_clusters=20, merge_remote=False, merge_thres=0.8, norm_thres=4000)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[38]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[15, 15])
ds.plot_domains(rotate=3, cmap=&#39;rainbow&#39;, z=0)
plt.axis(&#39;off&#39;)
plt.tight_layout()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_40_0.png" src="../_images/results_nb2.consensus_merfish_40_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[39]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>layer_annotations = ds.inferred_domains[ds.local_maxs]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>calls_gmcs = pd.read_csv(&quot;consensus_calls/charles/merfish_jeremy_pciseq_renee_eeshit_yilin_gabriele_consensus_df.csv&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>for cl in calls_gmcs.subclass.unique():
    if cl == &quot;L23_IT&quot;:
        calls_gmcs.subclass.loc[calls_gmcs.subclass == &quot;L23_IT&quot;] = &quot;L2/3 IT&quot;
    elif &quot;_&quot; in cl:
        calls_gmcs.subclass.loc[calls_gmcs.subclass == cl] = cl.replace(&quot;_&quot;, &quot; &quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/tmp/ipykernel_576/1845947111.py:5: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  calls_gmcs.subclass.loc[calls_gmcs.subclass == cl] = cl.replace(&#34;_&#34;, &#34; &#34;)
/tmp/ipykernel_576/1845947111.py:3: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  calls_gmcs.subclass.loc[calls_gmcs.subclass == &#34;L23_IT&#34;] = &#34;L2/3 IT&#34;
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>uniq_celltypes_gmcs = [cl for cl in cell_class_colors.keys() if cl in calls_gmcs.subclass.unique()]
centroids_gmcs = []
for cell_type in uniq_celltypes_gmcs:
    centroids_gmcs.append(np.mean(cell_by_gene_normalized[calls_gmcs.subclass == cell_type], axis=0))
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>centroids_scaled_gmcs = []
for cell_type in uniq_celltypes_gmcs:
    centroids_scaled_gmcs.append(np.mean(cell_by_gene_normalized_scaled[calls_gmcs.subclass == cell_type], axis=0))

sorted_cbg, sorted_genes = sort_genes(centroids_scaled_gmcs, cell_by_gene_normalized_scaled, ds.genes)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>cols = [cell_class_colors[ct] for ct in uniq_celltypes_gmcs]
plot_heatmap(sorted_cbg[:, ::-1], sorted_genes[::-1], calls_gmcs, uniq_celltypes_gmcs, cols, [20, 50]).savefig(&quot;merfish_heatmap_gmcs.pdf&quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_46_0.png" src="../_images/results_nb2.consensus_merfish_46_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from sklearn import preprocessing
fig, axes = plt.subplots(len(uniq_celltypes_gmcs), 1, figsize=[20, len(uniq_celltypes_gmcs)*2])
plt.subplots_adjust(hspace=0)
for idx, cell_type in enumerate(uniq_celltypes_gmcs):
    cl_vecs = cell_by_gene_normalized_scaled[calls_gmcs.subclass == cell_type]
    if len(cl_vecs) == 1:
        cl_vecs = np.array([cl_vecs[0], cl_vecs[0]])
    sns.violinplot(ax=axes[idx], data=cl_vecs, width=1)
    axes[idx].set_ylabel(cell_type)
    axes[idx].set_yticks([])
axes[idx].set_xticklabels(ds.genes, rotation=90)
pass
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_47_0.png" src="../_images/results_nb2.consensus_merfish_47_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>map_colors_gmcs = [cell_class_colors[ct.replace(&quot;_&quot;, &quot; &quot;).replace(&quot;L23&quot;, &quot;L2/3&quot;)] for ct in uniq_celltypes_gmcs]
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[150]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.map_celltypes(centroids_gmcs)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Generating cell-type map for centroid #0...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #1...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #2...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #3...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #4...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #5...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #6...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #7...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #8...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #9...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #10...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #11...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #12...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #13...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
Generating cell-type map for centroid #14...
Processing chunk (0/3)...
Processing chunk (1/3)...
Processing chunk (2/3)...
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[151]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>analysis.filter_celltypemaps(min_norm=0.6, min_r=0.6)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[152]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>plt.figure(figsize=[20, 20])
ds.plot_celltypes_map(rotate=3, colors=map_colors_gmcs)
plt.title(&quot;MERFISH - GMCS (SSAM)&quot;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[152]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 1.0, &#39;MERFISH - GMCS (SSAM)&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_51_1.png" src="../_images/results_nb2.consensus_merfish_51_1.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[44]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>closest_nwcs_clusters = []
closest_nwcs_clusters_r = []
closest_gmcs_clusters = []
closest_gmcs_clusters_r = []
for v in ds.normalized_vectors:
    corrs = [ssam.utils.corr(v, centroids_nwcs[i]) for i in range(len(centroids_nwcs))]
    idx = np.argmax(corrs)
    closest_nwcs_clusters.append(uniq_celltypes_nwcs[idx])
    closest_nwcs_clusters_r.append(corrs[idx])

    corrs = [ssam.utils.corr(v, centroids_gmcs[i]) for i in range(len(centroids_gmcs))]
    idx = np.argmax(corrs)
    closest_gmcs_clusters.append(uniq_celltypes_gmcs[idx])
    closest_gmcs_clusters_r.append(corrs[idx])

df = pd.DataFrame(ds.normalized_vectors, columns=ds.genes)
df.to_csv(&quot;merfish_ssam_localmax_expression.csv&quot;)

df = pd.DataFrame()
df[&#39;x&#39;] = ds.local_maxs[0]
df[&#39;y&#39;] = ds.local_maxs[1]
df[&#39;closest_consensus_nwcs_cluster&#39;] = closest_nwcs_clusters
df[&#39;closest_consensus_nwcs_cluster_r&#39;] = closest_nwcs_clusters_r
df[&#39;closest_consensus_gmcs_cluster&#39;] = closest_gmcs_clusters
df[&#39;closest_consensus_gmcs_cluster_r&#39;] = closest_gmcs_clusters_r
df[&#39;layer_annotations_nwcs&#39;] = layer_annotations

df.to_csv(&quot;merfish_ssam_localmax_metadata_with_layer.csv&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[155]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from scipy.spatial import ConvexHull

plt.figure(figsize=[20, 20])
plt.gca().set_facecolor(&#39;black&#39;)
good_ids = cell_by_gene.index.astype(int)
i = 0
for cid, sdf in locations.groupby(&quot;cell&quot;):
    if cid in good_ids:
        points = sdf.iloc[:, :2].to_numpy()
        hull = ConvexHull(points)
        plt.fill(points[hull.vertices, 0], points[hull.vertices, 1], cell_class_colors[calls_nwcs.subclass[i]], edgecolor=&quot;black&quot;, linewidth=0.5)
        i += 1
plt.xlim([0, ds.shape[0]])
plt.ylim([0, ds.shape[1]])
plt.gca().set_aspect(&#39;equal&#39;, adjustable=&#39;box&#39;)
plt.title(&quot;MERFISH - NWCS&quot;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[155]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 1.0, &#39;MERFISH - NWCS&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_53_1.png" src="../_images/results_nb2.consensus_merfish_53_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[154]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from scipy.spatial import ConvexHull

plt.figure(figsize=[20, 20])
plt.gca().set_facecolor(&#39;black&#39;)
good_ids = cell_by_gene.index.astype(int)
i = 0
for cid, sdf in locations.groupby(&quot;cell&quot;):
    if cid in good_ids:
        points = sdf.iloc[:, :2].to_numpy()
        hull = ConvexHull(points)
        plt.fill(points[hull.vertices, 0], points[hull.vertices, 1], cell_class_colors[calls_gmcs.subclass[i].replace(&quot;_&quot;, &quot; &quot;).replace(&quot;L23&quot;, &quot;L2/3&quot;)], edgecolor=&quot;black&quot;, linewidth=0.5)
        i += 1
plt.xlim([0, ds.shape[0]])
plt.ylim([0, ds.shape[1]])
plt.gca().set_aspect(&#39;equal&#39;, adjustable=&#39;box&#39;)
plt.title(&quot;MERFISH - GMCS&quot;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[154]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 1.0, &#39;MERFISH - GMCS&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/results_nb2.consensus_merfish_54_1.png" src="../_images/results_nb2.consensus_merfish_54_1.png" />
</div>
</div>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
        </div>
      </div>
    </div>

  <div id="footer">
    <p class="copyright">
      © 2022 SpaceTx consortium<br>
    </p>
  </div>

  </body>
</html>