Data Visualization
------------------

Spatial Transcriptomics data presents unique challenges for visualization. 

- Visualization of processed spatial transcriptomics data
   - Processed Spatial Transcriptomics data typically includes:
      - Spot Tables: Spatial locations for all mRNA spots detected in the experiment.
      - Cell by Gene Tables: Cell locations and the total mRNA counts associated with each cell.
      - Overview Image: spatial context for high-resolution data. For example, SpaceTx Consortium data includes high resolution data of mouse VISp and the overview image includes a DAPI image of the entire coronal section containing the VISp data.

  - Cell by Gene Visualization in Cytosplore
      - Boudewijn Lelieveldt and the Cytosplore Team have created a project-specific Cytosplore viewer for interactive exploration of SpaceTx cell by gene data
      - download link:
      - walk-through or screenshot
  - Segmentation-free data analysis and visualization
      - The heterogeneity of gene expression in tissue can be explored without cell segmentation.
      - The SSAM method (`Nature Communications volume 12, Article number: 3545 (2021) <https://www.nature.com/articles/s41467-021-23807-4>`_) applied to SpaceTx data creates an easily visualized landscape of gene expression that can highlight rare cell types or tissue organization. 

       .. image:: /data/ssam-spacetx.png
            :width: 800


- Image Data Visualization
   -  Unlike traditional biological imaging, spatial transcriptomics image data is often uninterpretible on its own. For example, each (round,channel) image from a MERFISH experiment typically contains diffraction-limited spots from all copies of 10s of different mRNA species. However, an image of DAPI from the same experiment could be very useful for visualizing tissue context for processed spatial transcriptomics data. 
   -  Raw image data for spatial transcriptomics is frequently at the terabyte scale and visualization requires comprehension of the spatial and (round,channel) organization of the images. 
   -  Modern bioimage visualization tools such as `Neuroglancer <https://github.com/google/neuroglancer>`_ and `Napari <https://napari.org/>`_ can be used to visualize spatial transcriptomics image data, but only to the extent that image data format information is interpretable by the visualization tool. Visualization tools will become more interoperable when and if the bioimaging community converges on formats and metadata standards for multiround fluorescence experiments. Large community standards projects like  `OME-NGFF <https://forum.image.sc/tag/ome-ngff>`_ as well as smaller, nascent projects like `useq-schema <https://github.com/tlambert03/useq-schema>`_ could form the basis of a metadata standard that could be applied to spatial transcriptomics microscopy data to simplify interactive visualization of large scale image data.

